### PR TITLE
CPLAT-6814 Update copyUnconsumedDomProps usage

### PR DESCRIPTION
**Pull request by sydneyjodon-wk**
## Motivation
  
Because of Component to Component2 update, `copyUnconsumedDomProps` and `copyUnconsumedProps` usages need to be updated:
- Update copyUnconsumedDomProps usage:
`- ..addProps(copyUnconsumedDomProps())`
`+ ..modifyProps(addUnconsumedDomProps)`
- Update copyUnconsumedProps usage:
`- ..addProps(copyUnconsumedProps())`
`+ ..modifyProps(addUnconsumedProps)`
## Changes
  
- Add suggestor that updates `copyUnconsumedDomProps` and `copyUnconsumedProps` usages.
- Write tests.
## Review
_[See CONTRIBUTING.md][contributing-review-types] for more details on review types (+1 / QA +1 / +10) and code review process._

  

Please review: 
@kealjones-wk @aaronlademann-wf @greglittlefield-wf @joebingham-wk 
### QA Checklist
- [ ] Tests were updated and provide good coverage of the changeset and other affected code
- [ ] Manual testing was performed if needed
    - [ ] Steps from PR author: 
        
      - [ ] CI passes
    - [ ] Anything falling under manual testing criteria [outlined in CONTRIBUTING.md][contributing-manual-testing]

## Merge Checklist
While we perform many automated checks before auto-merging, some manual checks are needed:
- [ ] A Client Platform member has reviewed these changes
- [ ] There are no unaddressed comments _- this check can be automated if reviewers use the "Request Changes" feature_
- [ ] _For release PRs -_ Version metadata in Rosie comment is correct

[contributing-review-types]: https://github.com/Workiva/over_react_codemod/blob/master/CONTRIBUTING.md#review-types
[contributing-manual-testing]: https://github.com/Workiva/over_react_codemod/blob/master/CONTRIBUTING.md#manual-testing-criteria

### CODE REVIEW

**Comment by aviary-wf** [comment]
## Security Insights
No security relevant content was detected by automated scans.

## Action Items
* Review PR for [security impact](https://wiki.atl.workiva.net/display/SECURITY/Development+Security+Review+Guidelines); comment "security review required" if needed or unsure
* Verify [`aviary.yaml`](https://dev.workiva.net/docs/teams/information-security/aviary#configuring-aviary) coverage of security relevant code
---
_Questions or Comments? Reach out on Slack: #support-infosec._

File: lib/src/component2_suggestors/copyunconsumeddomprops_migrator.dart [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+class CopyUnconsumedDomPropsMigrator extends GeneralizingAstVisitor
+    with AstVisitingSuggestorMixin
+    implements Suggestor {
+  CopyUnconsumedDomPropsMigrator();
+
+  @override
+  visitMethodInvocation(MethodInvocation node) {
+    super.visitMethodInvocation(node);
+
+    if (node.methodName.toString() == 'addProps') {
```
Comment (greglittlefield-wf) [inline lib/src/component2_suggestors/copyunconsumeddomprops_migrator.dart]:
```suggestion
    if (node.methodName.name == 'addProps') {
```

File: lib/src/component2_suggestors/copyunconsumeddomprops_migrator.dart [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+      var firstArg = node.argumentList.childEntities
+          .firstWhere((a) => a.toString() != '(');
```
Comment (greglittlefield-wf) [inline lib/src/component2_suggestors/copyunconsumeddomprops_migrator.dart]: I'd recommend using the AST API for this:
```dart
var firstArg = node.argumentList.arguments.first;
```

File: lib/src/component2_suggestors/copyunconsumeddomprops_migrator.dart [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+      if (firstArg.toString() == 'copyUnconsumedDomProps()' ||
```
Comment (greglittlefield-wf) [inline lib/src/component2_suggestors/copyunconsumeddomprops_migrator.dart]: See previous PR; should use `toSource`
```suggestion
      if (firstArg.toSource() == 'copyUnconsumedDomProps()' ||
```

File: lib/src/component2_suggestors/copyunconsumeddomprops_migrator.dart [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+          firstArg.toString() == 'copyUnconsumedProps()') {
```
Comment (greglittlefield-wf) [inline lib/src/component2_suggestors/copyunconsumeddomprops_migrator.dart]:
```suggestion
          firstArg.toSource() == 'copyUnconsumedProps()') {
```

File: lib/src/component2_suggestors/copyunconsumeddomprops_migrator.dart [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+        // Update argument.
+        yieldPatch(firstArg.offset, firstArg.end,
+            'addUnconsumed${firstArg.toString().contains('Dom') ? 'Dom' : ''}Props');
```
Comment (greglittlefield-wf) [inline lib/src/component2_suggestors/copyunconsumeddomprops_migrator.dart]:
```suggestion
            'addUnconsumed${firstArg.toSource().contains('Dom') ? 'Dom' : ''}Props');
```

File: lib/src/component2_suggestors/copyunconsumeddomprops_migrator.dart (hunk 2 above)
Comment (sydneyjodon-wk) [reply lib/src/component2_suggestors/copyunconsumeddomprops_migrator.dart]: That's way better! Thank you!

**Pull request merged**